[FLINK-17491] Translate Training page on project website

liying919 opened the pull request:
Translate Training page on project website.

Commit:
[FLINK-13683] Translate "Code Style - Component Guide" page into Chinese

This closes #247

yangjf2019 committed:
[FLINK-13343][docs-zh] Translate "Contribute Code" page into Chinese

This closes #268

wuchong committed:
Add Blog: "Flink SQL Demo: Building an End to End Streaming Application"

morsapaes committed:
Correcting Dawid's status.

This closes #337

zjffdu committed:
[blog] flink on zeppelin

klion26 committed:
[FLINK-17926] Singleton the command used to build website using Docker

Before we can build the website using `./docker/run.sh` and the command `docker run --rm --volume="$PWD:/srv/flink-web" --expose=4000 -p 4000:4000 -it ruby:2.5 bash -c 'cd /srv/flink-web && ./build.sh -p'` (which in the the page `Contribute to the Website`).

Now we remove the `./docker` directory, and using the command as the only official way to build Flink website.

Also add the command to README.md, so that we can easily find it.

This closes #342

zjffdu committed:
Apply suggestions from code review

address comment

Co-authored-by: MarkSfik <47176197+MarkSfik@users.noreply.github.com>
Co-authored-by: morsapaes <marta.paes.moreira@gmail.com>

yangqu committed:
[hotfix] Fix typo in reviewing-prs.md (#343)

hequn8128 committed:
Correct Hequn's family name in the blog

hequn8128 committed:
Rebuild website

zjffdu committed:
update blog for part-1

zjffdu committed:
[blog] flink on zeppelin - part2

Co-authored-by: morsapaes <marta.paes.moreira@gmail.com>

This closes #344.

liying919 committed:
[FLINK-17491] Translate Training page on project website

liying919 committed:
[FLINK-17491] Translate Training page on project website

tzulitai committed:
[release] Update downloads and doc links for StateFun 2.1.0

This closes #347.

morsapaes committed:
Adding a blogpost for the announcement of StateFun 2.1.

This closes #348.

rmetzger committed:
regenerate page

tzulitai committed:
Rebuild website

morsapaes committed:
[blog] Adding a Community Update blogpost for June.

This closes #349

sjwiesman committed:
[hotfix] Update docs style for creating cross links

sjwiesman committed:
rebuild website

klion26 commented on training.zh.md:
```diff
@@ -1,54 +1,52 @@
-The Apache Flink community maintains a self-paced training course that contains
-a set of lessons and hands-on exercises. This step-by-step introduction to Flink focuses
-on learning how to use the DataStream API to meet the needs of common, real-world use cases.
+Apache Flink 社区维护了一套自学培训课程，包括了一系列课程和上手练习。 这个对 Flink 的逐步介绍关注于学习如何使用 DataStream API 来满足常见的实际使用需求。
```
`这个对 Flink 的逐步介绍关注于学习如何使用 DataStream API 来满足常见的实际使用需求。` 这句话感觉有点拗口。能否修改一下呢

liying919 replied:
修改为了：
`这个对 Flink 的分步介绍重点在于学习如何使用 DataStream API 来满足常见的实际使用需求。`

klion26 commented on training.zh.md:
```diff
@@ -1,54 +1,52 @@
-                    <li>Sources and Sinks</li>
+                    <li>数据类型和序列化</li>
+                    <li>架构</li>
+                    <li>数据源和 Sink</li>
```
Source 和 sink 建议都继续翻译，和网站其他翻译保持一致

liying919 replied:
参考了[wiki](https://cwiki.apache.org/confluence/display/FLINK/Flink+Translation+Specifications) 说sink不做翻译，这边source我也不翻译了吧

klion26 commented on training.zh.md:
```diff
@@ -58,49 +56,49 @@ This training covers the fundamentals of Flink, including:
-                <li>Watermarks</li>
-                <li>Windows</li>
+                <li>事件时间处理</li>
+                <li>水印</li>
```
watermark 可以不翻译，可以参考 [wiki](https://cwiki.apache.org/confluence/display/FLINK/Flink+Translation+Specifications)

klion26 commented on training.zh.md:
```diff
@@ -58,49 +56,49 @@ This training covers the fundamentals of Flink, including:
-                <li>Exactly-once End-to-end</li>
+                <li>Checkpoints 和 Savepoints</li>
+                <li>精确一次与至少一次</li>
+                <li>精确一次的端到端</li>
```
`端到端的精确一次` 会更好一些吗？

klion26 reviewed (commented):
@liying919 thanks for your contribution, left some comments.

morsapaes committed:
Syncing the ZH version of the Docs Style Guide with recent changes and correcting an additional point.

sjwiesman committed:
rebuild website

rmetzger committed:
regenerate page

liying919 committed:
Update training.zh.md

klion26 commented on training.zh.md:
```diff
@@ -58,49 +57,49 @@ This training covers the fundamentals of Flink, including:
 
 <div style="margin-bottom: 400px;">
-<a href="{{site.DOCS_BASE_URL}}flink-docs-master/zh/training" target='_blank'>Apache Flink Training Course <small><span class="glyphicon glyphicon-new-window"></span></small> </a> 
+<a href="{{site.DOCS_BASE_URL}}flink-docs-master/zh/training" target='_blank'>Apache Flink 培训课程 <small><span class="glyphicon glyphicon-new-window"></span></small> </a> 
```
这个链接会 404，可以参考英文版的继续下修改。

liying919 replied:
这里不太明白，英文版中的链接是
`href="{{site.DOCS_BASE_URL}}flink-docs-master/training" `
只差个 /zh/

klion26 replied:
英文版实际上是有跳转的（这个应该是在项目中某个地方设置了跳转，我暂时没有找到），最终的链接是 `https://ci.apache.org/projects/flink/flink-docs-master/learn-flink/index.html`

liying919 replied:
@alpinegizmo Hi David, @klion26 is helping review this PR. We found the link _{{site.DOCS_BASE_URL}}flink-docs-master/training_ in training.md auto jumps to _https://ci.apache.org/projects/flink/flink-docs-master/learn-flink/index.html_ . Do you know where we can set the same jump config for Chinese version?

klion26 replied:
I'm not very sure if we put the indeed URL(`https://ci.apache.org/projects/flink/flink-docs-master/learn-flink/index.html`) here is ok, currently, English version will redirect to it.

alpinegizmo replied:
@liying919 @klion26 That redirect is configured in docs/redirects/training.md.

alpinegizmo replied:
https://github.com/apache/flink-web/pull/357 updated this link to point to {{site.DOCS_BASE_URL}}flink-docs-stable/zh/learn-flink

klion26 commented on training.zh.md:
```diff
@@ -1,54 +1,53 @@
-                <li>Connected Streams</li>
+                <li>转换</li>
+                <li>有状态的流处理</li>
+                <li>关联的流</li>
```
这个如果改成`流的关联` 会好一些吗？

liying919 replied:
已修改

klion26 reviewed (commented):
@liying919 thanks for the work. left another two comments. please take another look, thanks.

fhueske committed:
Rebuild website

azagrebin committed:
[FLINK-18309] Recommend avoiding uppercase to emphasise statements in doc style

This closes #351.

azagrebin committed:
rebuild website

rmetzger committed:
regenerate website

Commit:
Update training.zh.md

carp84 committed:
Rebuild website

morsapaes committed:
Adding a release blogpost for Flink 1.11.0.

azagrebin committed:
Add a code contribution section about how to look for what to contribute

This closes #353.

azagrebin committed:
Rebuild website

afedulov committed:
Link blogposts

This closes #354.

wuchong committed:
Update contents to 1.11.0

pnowojski committed:
Add download links for the 1.11.0 release

pnowojski committed:
Update quickstart scripts for 1.11.0 release

pnowojski committed:
Remove download links for the 1.9.x release

pnowojski committed:
Rebuild website

Commit:
Rebuild website

pnowojski committed:
Rebuild website

pnowojski committed:
fixup! Rebuild website

alpinegizmo committed:
[hotfix] update links to point to training in stable docs

alpinegizmo committed:
[hotfix] update links to point to training in stable docs

wkodate committed:
[hotfix] Fix broken link on flink 1.11.0 release blog

sjwiesman committed:
rebuild website

klion26 commented:
hi @liying919,  what's the status of current PR?

liying919 commented:
> 
> 
> hi @liying919, what's the status of current PR?



> 
> 
> hi @liying919, what's the status of current PR?

The redirect config is in Apache/Flink.  @alpinegizmo Should we add the same redirect to https://ci.apache.org/projects/flink/flink-docs-master/zh/learn-flink/index.html in Project Flink?

alpinegizmo commented:
@liying919 We needed that redirect in English because the old URL had already been published in a few places, and we didn't want to break those links. I don't know if the Chinese link has been published anywhere, but maybe not. 

Also, now that the training is present in the stable docs, we should not be linking to the training in the master docs.

kl0u committed:
Application Mode blogpost

morsapaes committed:
Making the blogpost compilable and adding a few adjustments.

dianfu committed:
Add Apache Flink release 1.11.1

This closes #359.

sjwiesman committed:
rebuild website

sjwiesman committed:
rebuild site

sjwiesman committed:
rebuild site

Commit:
[FLINK-17491] Translate Training page on project website

liying919 commented:
@klion26 Please refer to the new commit.

klion26 commented:
@liying919 thanks for the work, the change LGTM, but there is some conflict, could you please rebase the master to resolve the conflict?

dawidwys committed:
Catalogs blogpost

dawidwys committed:
Rebuild website

liying919 commented:
@klion26 The default branch has been rebased.

dianfu committed:
Rebuild website

klion26 commented:
@liying919 thanks for the work. seems there is something wrong(now there are 258 files touched), could you please take another look.

afedulov committed:
Add Blogbost: Advanced Flink Application Patterns: Custom Window Processing

This closes #362.

dawidwys committed:
Rebuild website

dawidwys committed:
Fix code block in catalogs blogpost

dawidwys committed:
Rebuild website

wuchong committed:
Apply suggestions from code review

Co-authored-by: morsapaes <marta.paes.moreira@gmail.com>

wuchong committed:
rename file and images to 07-28

wuchong committed:
Update 1.11.0 to 1.11

sunjincheng121 committed:
Add blog post for Pandas support in PyFlink.

This closes #364.

sjwiesman committed:
rebuild website

morsapaes committed:
Adding a Community Update blogpost for July.

sjwiesman committed:
fix links and rebuild

NicoK committed:
Rebuild website

NicoK committed:
Rebuild website

rmetzger committed:
Rebuild website

This closes #pr363

klion26 commented:
@liying919 kindly ping.

KarmaGYZ committed:
Add blog post for External Resources

This closes #365.

dianfu committed:
Rebuild website

dianfu committed:
Update the blog date of 'Pandas support in PyFlink'

dianfu committed:
Rebuild website

tillrohrmann committed:
Rebuild website

alpinegizmo committed:
Add danderson to committer list

alpinegizmo committed:
Rebuild site

XComp committed:
[FLINK-18963][site] Added Copyright information to common code style guidelines.

This closes #368.

tillrohrmann committed:
Rebuild site

rmetzger committed:
[blog] Add post about flink on docker

This closes #370

morsapaes committed:
Adding Igal's blogpost on StateFun.

morsapaes committed:
Pointing to the docs master instead.

sjwiesman committed:
rebuild site

rmetzger committed:
rebuild page

zhuzhurk committed:
Add Apache Flink release 1.10.2

sjwiesman committed:
[hotfix] Add makefile

zhuzhurk committed:
Rebuild site

dianfu committed:
Update status for Dian to PMC in community web page

azagrebin committed:
Add blog post: Memory Management improvements for Flink’s JobManager in Apache Flink 1.11

This closes #373.

XComp committed:
[hotfix] Introduced new parameter to enable futures during build.

Additional changes:
- Updated Jekyll's excludes (removing obsolete entries and added README.md [+ docker-build.sh])
- Replaced Makefile by docker-build.sh to make parameter passing easier

This closes #374

morsapaes committed:
Adding a Community Update blogpost for August(ish).

azagrebin committed:
Rebuild site

sjwiesman committed:
[hotfix] fix makefile

rmetzger committed:
regenerate website

This closes #375

afedulov committed:
[hotfix] add links to Advanced Application Patterns Vol. 3

sjwiesman committed:
rebuild site

xintongsong committed:
Add xtsong to the committer list.

This closes #378

xintongsong committed:
Rebuild site.

Shawn-Hx committed:
[hotfix] Fix typo: commiter --> committer (#372)

zhuzhurk committed:
Add Apache Flink release 1.11.2

zhuzhurk committed:
Rebuild site

tzulitai committed:
[release] [statefun] Announcement post for v2.2.0

tzulitai committed:
[release] [statefun] Add v2.2.0 to Downloads page

This closes #379.

tzulitai committed:
Rebuild website

tzulitai committed:
[hotfix] Release StateFun 2.0.0 from Downloads page

tzulitai committed:
Rebuild website

rmetzger committed:
add flink-shaded 12.0 release

rmetzger committed:
rebuild website

tzulitai committed:
Add blog "Stateful Functions Internals: Behind the scenes of Stateful Serverless"

tzulitai committed:
Rebuild website

tzulitai committed:
Rebuild website

tzulitai committed:
Fix typo and rebuild website

Commit:
Add blog "From Aligned- to Unaligned Checkpoints - Part 1: Checkpoints, Alignment, and Backpressure"

This closes #387

rmetzger committed:
regenerate website

carp84 committed:
Update status for Yu to PMC in community web page

tzulitai committed:
Add release announcement for StateFun 2.2.1

tzulitai committed:
Update Downloads page config for StateFun 2.2.1

tzulitai committed:
Rebuild website

dawidwys committed:
[FLINK-20093] Link to SQL components download page (#391)

wangxlong committed:
[FLINK-20129] Link to zh SQL components download page (#393)

liying919 commented:
> @liying919 thanks for the work, the change LGTM, but there is some conflict, could you please rebase the master to resolve the conflict?

Pls refer to the new PR.
https://github.com/apache/flink-web/pull/395
